systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalclients/KafkaClient.java

```diff
@@ -68,17 +73,24 @@ public void close() {
     @Override
     public Future<Integer> sendMessagesTls(String topicName, String namespace, String clusterName, String kafkaUsername, int messageCount, String securityProtocol) {
         String clientName = "sender-ssl" + clusterName;
+        vertx = Vertx.vertx();
         CompletableFuture<Integer> resultPromise = new CompletableFuture<>();
 
         IntPredicate msgCntPredicate = x -> x == messageCount;
 
-        vertx.deployVerticle(new Producer(KafkaClientProperties.createProducerProperties(namespace, clusterName, kafkaUsername, securityProtocol, EClientType.BASIC, null), resultPromise, msgCntPredicate, topicName, clientName));
+        String clientSecretName = getCaCertName() == null ? KafkaResource.getKafkaExternalListenerCaCertName(namespace, clusterName) : getCaCertName();
```

Comment:
Just a detail, by java convention there is no need to use getters unless we perfoming some control logic inside getters. Change `getCaCertName()` --> caCertName or this.caCertName

- The reason for hiding is mainly to hide the implementation (and there's no need to hide the implementation from the class itself)